### DISCUSSION TITLE
Práctica Js (ayuda)

### DISCUSSION THREAD

**Discussion by GloriaCoralCerecedo** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#448] [Q&A]
Hola, tengo duda de cómo poner los tipos, estadísticas y movimientos. 

Si me podrían proporcionar alguna idea. Eh visto varios pero trato de comprenderlos.

Lo que imagino

En el del tipo porque aveces hay Pokémon de 1 tipo y otros de 2 tipos, se podría hacer un for igual ví otras maneras pero estoy trabada ahí 

Me imagino igual que para las estadísticas y movimientos se podría poner otro fetch

**Comment by hackit92** [comment]
Pues yo te diría que en una función incluyas todas las sub funciones, así podrías aprovechar un solo fetch para obtener la información de todas. Por otro lado puedes utilizar una función que concatene todos los tipos de pokemon y si solo tiene 1 mande el valor solo para evitar que diga undefined

**Reply by GloriaCoralCerecedo** [reply]
Si voy a checar eso, eso es lo que quiero hacer y masomenos vi como. Gracias !

**Comment by Drako9159** [comment]
Pensaba hacer un menú desplegable para los movimientos o algo así, pero no iba con el estilo de pokédex que estaba haciendo, así que simplemente agregué un botón que suma en 1 al valor del movimiento e ir cambiando, sabiendo que tienen hasta 100 movimientos. https://drako9159.github.io/JS/dex.html

**Reply by GloriaCoralCerecedo** [reply]
Gracias! Si es que sin varios, igual que por ejemplo te den 5 movimientos y con un botón vas mostrando el resto. Cómo Next

**Comment by PlayerOneV** [comment]
Saludos para los tipos que puede tener un Pokémon (por lo que se máx. 2) lo que yo hice fue:

1.- Recibir la matriz de los tipos
2.- Guardar el primer tipo que tenga el Pokémon (índice 0)
3.- Comprobar si existe un segundo tipo (utilizando el índice 1) usando un if
4.- Ahora si existe el segundo tipo lo envió como parametro junto al primero a la función que hace que se muestre en el front
5.- Si no existe solamente envió el tipo 1 a la función

Para que el parametro tipo2 de la función sea opcional lo que se hace es inicializarlo con un valor vacío (tipo1, tipo2="").

Para finalizar tenia que ocultar el componente del front HTML encargado de mostrar el tipo2 dependiendo de si existía o no para eso cree una clase hide en css que tenia display:none para ocultar el elemento en donde sea agregada esa clase. Para agregar o remover esta clase dependiendo se si existe un tipo2 se utiliza la constante en donde se guarda el elemento HTML encontrado con el Id de esta forma:
* constante.classList.add/remove("Nombre de la clase que se agregara o removerá").

No es necesario utilizar otro fetch para las estadísticas y movimientos, para que no se te acumule mucho el código es preferible utilizar funciones y las llamarías dentro del fetch una vez hayas recuperado los valores que necesitas enviándolos como parámetros.

Espero haberte ayudado, te comparto una parte del codigo sobre como recupere los valores de la matriz de types.
`

         let type = info.types;    // primero obtenemos el array de tipos

         let type1 = type[0]["type"]["name"];    //Guardamos el primer tipo
        
        if (type[1]) { //Si el pokemon tiene un segundo tipo lo guardamos y enviamos como parametro para que se muestre
            let type2 = type[1]["type"]["name"];
            pokeType(type1, type2);

        } else { //Si el pokemon no tiene un segundo tipo solamente enviamos el primero
            pokeType(type1);
        }
`
